src/platform/packages/shared/kbn-alerts-ui-shared/src/common/types/action_types.ts

```diff
@@ -28,6 +28,7 @@ export type ConnectorValidationFunc = () => Promise<ConnectorValidationError | v
 export interface ActionConnectorFieldsProps {
   readOnly: boolean;
   isEdit: boolean;
+  isServerless?: boolean;
```

Comment:
I recall we shouldn’t be enabling or disabling features based on the serverless flag - it should be a more atomic approach. In our serverless config files, we have `xpack.ml.nlp.modelDeployment.allowStaticAllocations`.

Should we rely on `allowStaticAllocations` instead and pass it as a prop here?

Reply:
Good reminded - updated to `enforceAdaptiveAllocations` here ebff5bc61d7b5212ac76edd658fff6c6f2a44c42